src/data/nat/factorization.lean

```diff
@@ -52,6 +52,10 @@ by simp only [support_factorization, list.mem_to_finset]
 lemma factorization_eq_zero_iff (n : ℕ) : n.factorization = 0 ↔ n = 0 ∨ n = 1 :=
 by simp [factorization, add_equiv.map_eq_zero_iff, multiset.coe_eq_zero]
 
+@[simp] lemma factorization_mul_of_pos {a b : ℕ} (ha : 0 < a) (hb : 0 < b) :
```

Comment:
Ok.  Why is `a ≠ 0` preferable to `0 < a`?  Should I make the corresponding change to `perm_factors_mul_of_pos` and `count_factors_mul_of_pos` in `data/nat/prime`?

Reply:
That's something we've decided. Lemmas about naturals should take `a ≠ 0` as inputs and `0 < a` as outputs.

Reply:
This was basically before `bot_lt`, because `le.ne'` is quite short and the old was something like `nat.pos_of_ne_zero`. Now with `bot_lt`, I don't mind as much, but I guess it's still nice